### PR TITLE
adds `HttpStatusException` for non-200 responses

### PR DESCRIPTION
**Pull request by barend**
This exception type exposes the HTTP status code and response body for programmatic access. It is returned for HTTP status codes other than 200.

### CODE REVIEW

File: src/main/scala/ing/wbaa/druid/client/DruidClient.scala [hunk 1]
```diff
@@ -132,17 +132,17 @@ trait DruidResponseHandler {
...
     if (response.status != StatusCodes.OK) {
       body.flatMap { b =>
         Future.failed(
-          new Exception(s"Got unexpected response (with status ${response.status}) from Druid: $b")
+          new HttpStatusException(response.status, Some(b))
         )
       }
     } else {
       body
+        .map(e => e.data.decodeString("UTF-8"))
```
Comment (Fokko) [inline src/main/scala/ing/wbaa/druid/client/DruidClient.scala]: I prefer `.map(_.data.decodeString("UTF-8"))` 😃

File: src/test/scala/ing/wbaa/druid/DruidClientSpec.scala [hunk 2]
```diff
@@ -57,7 +59,34 @@ class DruidClientSpec extends WordSpec with Matchers with ScalaFutures {
...
+        case exception: HttpStatusException =>
+          exception.status shouldBe StatusCode.int2StatusCode(500)
+          exception.entity match {
+            case Some(entity) => entity.isKnownEmpty() shouldBe true
+            case _            => fail("expected empty entity, got empty option")
+          }
+        case response => fail(s"expected HttpStatusException, got $response")
+      }
+
+      client.shutdown().futureValue
```
Comment (Fokko) [inline src/test/scala/ing/wbaa/druid/DruidClientSpec.scala]: In the future we should have some kind of fixture for this. But that's unrelated to this PR.

File: src/main/scala/ing/wbaa/druid/client/DruidAdvancedHttpClient.scala [hunk 3]
```diff
@@ -283,7 +284,10 @@ object DruidAdvancedHttpClient extends DruidClientBuilder {
     val retryDelay =
       FiniteDuration(clientConfig.getDuration(Parameters.QueryRetryDelay).getNano, NANOSECONDS)
 
-    val brokerFlows = createConnectionFlows(druidConfig.hosts, druidConfig.secure, poolConfig)
+    val brokerFlows = createConnectionFlows(druidConfig.hosts,
+                                            druidConfig.secure,
+                                            druidConfig.responseParsingTimeout,
```
Comment (Fokko) [inline src/main/scala/ing/wbaa/druid/client/DruidAdvancedHttpClient.scala]: Interesting, the `responseParsingTimeout` was already in the config, but we weren't using it 🤔

File: src/main/scala/ing/wbaa/druid/client/DruidClient.scala (hunk 1 above)
Comment (barend) [reply src/main/scala/ing/wbaa/druid/client/DruidClient.scala]: Want me to update the PR?

Comment (Fokko) [reply src/main/scala/ing/wbaa/druid/client/DruidClient.scala]: No, we don't have any written code conventions, so it is fine like this as well.

**Pull request merged**